src/shared/backend/graph/firmware/cloudbuild.ts

```diff
@@ -1,22 +1,114 @@
+import config from "shared/backend/config";
 import { createBuilder } from "shared/backend/utils/builder";
 
 const builder = createBuilder();
 
-const Simple = builder.simpleObject("Simpler", {
+const Release = builder.simpleObject("Release", {
   fields: (t) => ({
-    bloup: t.string(),
+    id: t.string(),
+    name: t.string(),
+    description: t.string(),
+    excludeTargets: t.stringList(),
+    sha: t.string(),
+    isPrerelease: t.boolean(),
+  }),
+});
+
+const Flags = builder.simpleObject("Flags", {
+  fields: (t) => ({
+    key: t.string(),
+    value: t.stringList(),
+  }),
+});
+
+// flags: Record<string, string[]>
+const Tag = builder.simpleObject("Tag", {
+  fields: (t) => ({
+    flags: t.field({ type: Flags }),
+  }),
+});
+
+const TargetFlag = builder.simpleObject("TargetFlag", {
+  fields: (t) => ({
+    values: t.stringList(),
+  }),
+});
+
+const Target = builder.simpleObject("Target", {
+  fields: (t) => ({
+    description: t.string(),
+    tags: t.stringList(),
+  }),
+});
+
+const TargetFlags = builder.simpleObject("TargetFlags", {
+  fields: (t) => ({
+    key: t.string(),
+    value: t.field({ type: TargetFlag }),
+  }),
+});
+
+const Tags = builder.simpleObject("Tags", {
+  fields: (t) => ({
+    key: t.string(),
+    value: t.field({ type: Tag }),
+  }),
+});
+
+const Targets = builder.simpleObject("Targets", {
+  fields: (t) => ({
+    key: t.string(),
+    value: t.field({ type: Target }),
+  }),
+});
+
+// releases: Record<string, Release>;
+// flags: Record<string, TargetFlag>;
+// tags: Record<string, Tag>;
+// targets: Record<string, Target>;
+const CloudTargets = builder.simpleObject("CloudTargets", {
+  fields: (t) => ({
+    releases: t.field({ type: [Release] }),
+    flags: t.field({ type: [TargetFlags] }),
+    tags: t.field({ type: [Tags] }),
+    targets: t.field({ type: [Targets] }),
   }),
 });
 
 builder.queryType({
   fields: (t) => ({
-    simple: t.field({
-      type: Simple,
-      args: {
-        name: t.arg.string(),
-      },
-      resolve: (_, { name }) => {
-        return { bloup: name ?? "bloup bloup" };
+    cloudFirmwares: t.field({
+      type: CloudTargets,
+      resolve: async (_, __, { cloudbuild, github }) => {
+        const targets = await cloudbuild.fetchTargets();
```

Comment:
IMO you could make a resolver for this here and then exclude the from the firmware releases on the frontend. Up to you though, main issue I see is that you'd had to re-implement lots of the existing endpoints in the github resolvers :/

Reply:
One issue for me was the loading state, I would have to synchronize it with both queries. 
And I have the advantage of having more control on the data without updating the github schema.

Reply:
You can load as many resolvers as you like with a single GQL query.
For example
```
query Something {
   something(...) {
     id
     ...
   }
   somethingElse {
      ...
    }
```